README.md

```diff
@@ -435,6 +355,347 @@ One App can be started via docker or if built from source by running `node lib/s
+    /* All Modules */
+    appCompatibility,
+  };
+}
+```
+
+**Contents**
+* [provideStateConfig](#providestateconfig)
+* [csp](#csp)
+* [corsOrigins](#corsorigins)
+* [configureRequestLog](#configurerequestlog)
+* [extendSafeRequestRestrictedAttributes](#extendsaferequestrestrictedattributes)
+* [createSsrFetch](#createssrfetch)
+* [validateStateConfig](#validatestateconfig)
+* [requiredSafeRequestRestrictedAttributes](#requiredsaferequestrestrictedattributes)
+* [appCompatibility](#appcompatibility)
+
+##### `provideStateConfig`
+**Module Type**
+* ✅ Root Module 
+* 🚫 Child Module  
+
+**Shape**
+```js
+Module.appConfig = {
+  provideStateConfig: {
+    server: {
+      [settingName]: {
+        [environmentLevel]: String,
+      },
+    },
+    client: {
+      [settingName]: {
+        [environmentLevel]: String,
+      },
+    },
+  },
+};
+```
+
+The `provideStateConfig` directive is useful for supplying string-based key value settings per runtime (e.g. `client` or `server`) and per `environmentLevel` (e.g. QA, Prod, etc).
+
+ These values are injected into the exported React Components in *every* loaded Module:
+
+```js
+function MyModule({ config }) {
+  // Returns JSX
+}
+
+MyModule.propTypes = {
+  config: PropTypes.shape({
+    [settingName]: PropTypes.string,
+  }),
+};
+
+MyModule.appConfig = {
+  provideStateConfig,
+};
+
+export default MyModule;
+```
+
+**📘 More Information**
+* Example: [Frank Lloyd Root's `appConfig`](https://github.com/americanexpress/one-app/blob/master/prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/config.js)
+* [`validateStateConfig`](#validatestateconfig)
+
+##### `csp`
+**Module Type**
+* ✅ Root Module 
+* 🚫 Child Module  
+
+**Shape**
+```js
+RootModule.appConfig = {
+  csp: String,
+};
+```
+
+The `csp` static `String` should be a valid [Content Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) for your application which will be passed on to the HTML markup rendered by the Browser.
+
+> We recommend using something like [content-security-policy-builder](https://www.npmjs.com/package/content-security-policy-builder) to create your CSP string.
+
+**📘 More Information**
+* Example: [Frank Lloyd Root's CSP](./prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/csp.js)
+* [content-security-policy-builder](https://www.npmjs.com/package/content-security-policy-builder)
+
+##### `corsOrigins`
+**Module Type**
+* ✅ Root Module 
+* 🚫 Child Module  
+
+**Shape**
+```js
+Module.appConfig = {
+  corsOrigins: [String],
+};
+```
+
+The `corsOrigins` directive accepts an array of `String` URL origin domains.
+This will allow requests from those origins to make POST requests to the server.
+
+In practice, this allows POST requests from given origins to return partially rendered Modules.
+
+**📘 More Information**
+* [Frank Lloyd Root's `appConfig`](https://github.com/americanexpress/one-app/blob/master/prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/config.js)
```

Comment:
Can we use relative paths for this and all the other links to the sample modules? Does that work 🤔?

Reply:
Looks like we can 😄 https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-readmes#relative-links-and-image-paths-in-readme-files

Reply:
Fixed in 7bc9f58